connectors/src/connectors/zendesk/temporal/activities.ts

```diff
@@ -432,6 +447,74 @@ export async function syncZendeskTicketBatchActivity({
   };
 }
 
+/**
+ * This activity is responsible for syncing the next batch of recently updated articles to process.
+ * It is based on the incremental endpoint, which returns a diff.
+ */
+export async function syncZendeskTicketUpdateBatchActivity({
+  connectorId,
+  brandId,
+  currentSyncDateMs,
+  cursor,
+}: {
+  connectorId: ModelId;
+  brandId: number;
+  currentSyncDateMs: number;
+  cursor: string | null;
+}): Promise<{ hasMore: boolean; afterCursor: string | null }> {
+  const connector = await _getZendeskConnectorOrRaise(connectorId);
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const loggerArgs = {
+    workspaceId: dataSourceConfig.workspaceId,
+    connectorId,
+    provider: "zendesk",
+    dataSourceId: dataSourceConfig.dataSourceId,
+  };
+
+  const { accessToken, subdomain } = await getZendeskSubdomainAndAccessToken(
+    connector.connectionId
+  );
+  const zendeskApiClient = createZendeskClient({ accessToken, subdomain });
+  const brandSubdomain = await changeZendeskClientSubdomain(zendeskApiClient, {
+    connectorId,
+    brandId,
+  });
+
+  const startTime = Math.floor(currentSyncDateMs / 1000) - 60 * 5; // 5 min ago, previous scheduled execution
```

Comment:
Question here: what if we pause the connector? what if a connector takes a bit longer than 5mn to finish? What if 

In order to ensure we never miss minutes or even seconds, would this code work?

other connectors (gdrive, microsoft), store a cursor.
Otherwise, we could e.g. get the start of the last successful execution?

Reply:
this code would definitely break if any of the cases you mentioned.
The cursors provided by Zendesk's API have a short lifespan, I'll opt for storing the start of the last successful execution and using this as the next start date.

Reply:
Note: if a ticket is updated on Zendesk between the lastSyncSuccessfulTime and the time we fetch the tickets, it will be considered again on the next sync but this is not an issue because the ticket sync itself is idempotent

Reply:
Those tickets would be missing indeed, I misunderstood the "last sync successful" as the start of the last successful sync instead of its end 🤦

Reply:
I'll store the start date of the last successful sync in a new column of the table `zendesk_configurations` then and use it as the start date of the fetch, sounds reasonable @philipperolet?

Reply:
As per IRL, following the footsteps of Microsoft / Gdrive, a table with workspace / timestampCursor (named like this, representing the last successful sync start) is cleanest